src/box/replication.cc

```diff
@@ -996,6 +997,12 @@ bootstrap_leader_is_connected(struct applier **appliers, int count)
 			return true;
 		}
 	}
+	/*
```

Comment:
Nit: we probably don't want to check appliers at all in `supervised`. The result will anyway depend on `is_supervised_bootstrap_leader`. Should be moved to the beginning of the function

Reply:
Seems valid. Moved the check:

```diff
diff --git a/src/box/replication.cc b/src/box/replication.cc
index cdbe424a95..627e749651 100644
--- a/src/box/replication.cc
+++ b/src/box/replication.cc
@@ -990,6 +990,14 @@ applier_is_bootstrap_leader(const struct applier *applier)
 static bool
 bootstrap_leader_is_connected(struct applier **appliers, int count)
 {
+       /*
+        * In the supervised strategy it is possible that we know
+        * locally that the given instance is a bootstrap leader.
+        */
+       if (bootstrap_strategy == BOOTSTRAP_STRATEGY_SUPERVISED &&
+           is_supervised_bootstrap_leader)
+               return true;
+
        for (int i = 0; i < count; i++) {
                struct applier *applier = appliers[i];
                if (applier->state == APPLIER_CONNECTED &&
@@ -997,12 +1005,6 @@ bootstrap_leader_is_connected(struct applier **appliers, int count)
                        return true;
                }
        }
-       /*
-        * In the supervised strategy it is possible that we know
-        * locally that the given instance is a bootstrap leader.
-        */
-       if (bootstrap_strategy == BOOTSTRAP_STRATEGY_SUPERVISED)
-               return is_supervised_bootstrap_leader;
        return false;
 }
```